lib/solarwinds_apm/sampling/http_sampler.rb

```diff
@@ -36,27 +56,26 @@ def hostname
 
     def fetch_with_timeout(url, timeout_seconds = nil)
       uri = url
+      timeout = timeout_seconds || REQUEST_TIMEOUT
       response = nil
 
-      thread = Thread.new do
+      begin
         ::OpenTelemetry::Common::Utilities.untraced do
-          Net::HTTP.start(uri.host, uri.port, use_ssl: uri.scheme == 'https') do |http|
+          Net::HTTP.start(uri.host, uri.port,
+                          use_ssl: uri.scheme == 'https',
+                          open_timeout: timeout,
+                          read_timeout: timeout) do |http|
```

Comment:
if timeout is 10 seconds, does this mean we end up waiting 20 seconds (10 for open, 10 for read) to actually time out?

also just noticed a typo `GET_SETTING_DURAION` missing T